python/lsst/meas/astrom/match_probabilistic_task.py

```diff
@@ -71,17 +72,21 @@ def _apply_select_bool(
             The final selection array.
 
         """
+        is_pd = isinstance(catalog, pd.DataFrame)
         select_additional = (len(columns_true) + len(columns_false)) > 0
         if select_additional:
             if selection is None:
                 selection = np.ones(len(catalog), dtype=bool)
             for column in columns_true:
                 # This is intended for boolean columns, so the behaviour for non-boolean is not obvious
                 # More config options and/or using a ConfigurableActionField might be best
-                values = catalog[column].values
+                values = catalog[column]
+                if is_pd:
+                    values = values.values
                 selection &= (np.isfinite(values) & (values != 0))
             for column in columns_false:
-                selection &= (catalog[column].values == 0)
+                values = catalog[column]
+                selection &= ((values.values if is_pd else values) == 0)
```

Comment:
This could be `selection &= ((catalog[column] if not is_pd else catalog[column].values) == 0)` though that is a mouthful.